### ISSUE TITLE
Inconsistent usage of "·" and "∙"

### ISSUE THREAD

**Issue by VojtechStep** [UniMath/agda-unimath#464]
The first middle-dot is · ("MIDDLE DOT", codepoint #xb7), and it's used in left and right whiskering (`_·l_` and `_·r_` in `foundation-core.homotopies`).
The other one is ∙ ("BULLET OPERATOR", codepoint #x2219), and it's used in the same file for homotopy concatenation, just a few lines above left and right whiskering.

To my knowledge, the middle dot can't even be typed using agda-input.

From a cursory look through the output of greping for MIDDLE DOT, it's only used for these two whiskering operators and some prose in groups and rings. Perhaps these occurences should be replaced by the other operator?

**Comment by fredrik-bakke** [comment]
Are you unable to type this symbol using the sequence `cdot`?

**Comment by fredrik-bakke** [comment]
To my understanding, the difference in the symbols is used precisely to contrast whiskering from path concatenation.

**Comment by VojtechStep** [comment]
> Are you unable to type this symbol using the sequence `cdot`?

I'm not at my computer anymore, but I tried looking through all the symbols I could type with `\.`, and this wasn't one of them, `C-u C-x =` told me that I could only write it with `C-x 8 RET`, and I tried searching in agda-input.el, but it wasn't there; though now I see that Agda input inherits from TeX, so it's completely possible that I just didn't try hard enough.

And if the distinction is intentional, then maybe it could be written down somewhere? I'm not sure if it's the font I use (JetBrains Mono), or my Emacs configuration, but these two symbols look identical to me (I now see that they have different size on my phone).

**Comment by fredrik-bakke** [comment]
Well, that's definitively an issue; these symbols should not look the same. I would suggest changing your font.
The website uses the following series of fonts in prioritized order to display agda code:

```css
font-family: Menlo, Source Code Pro, Consolas, Monaco, Lucida Console, Liberation Mono, DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace, serif;
```
so you can try one of these.

The documentation can definitely be added. This is an unfortunate experience to have.

**Comment by fredrik-bakke** [comment]
I have created a pull request addressing this issue. Please take a look, and feel free to suggest further changes that can help mitigate this.

**Comment by VojtechStep** [comment]
> The website uses the following series of fonts in prioritized order (decreasing) to display agda code:
> 
> ```css
> font-family: Menlo, Source Code Pro, Consolas, Monaco, Lucida Console, Liberation Mono, DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace, serif;
> ```
> so you can try one of these.

Actually the two symbols look the same for me even on the website, see the attached screenshots. I'll check how it looks on my computer later in the afternoon.

I checked how the two symbols look in various contexts, and these are my results:
- My editor: indistinguishable (JetBrains Mono), I don't have a screenshot at the moment
- Website on mobile: indistinguishable (not sure which font is picked) [1][2]
- GitHub website on mobile: _barely_ distinguishable, I suspect that's mostly the work of different padding around the dots inside the quotation marks [3]
- GitHub application on mobile: indistinguishable [4]
- Discord bot message: clear difference [5]

![1](https://user-images.githubusercontent.com/15523887/219949217-5a5aece1-39c3-42d9-a869-098ef8b6d59b.png)
![2](https://user-images.githubusercontent.com/15523887/219949229-be3158c1-6955-4702-a178-52d2c0d3e57b.png)
![3](https://user-images.githubusercontent.com/15523887/219949428-32d2b163-6926-4cd4-ba56-2ab368a99923.png)
![4](https://user-images.githubusercontent.com/15523887/219949498-4677cd37-5241-4280-9aea-adb7acae4623.png)
![5](https://user-images.githubusercontent.com/15523887/219949558-8cdc45d1-afa2-47ae-a6c2-7765fda3fae0.png)

**Comment by fredrik-bakke** [comment]
Thank you for the additional information!
I am guessing this is due to your devices not having the most prioritized fonts, and whichever one they are falling back to is not good. Hopefully this can be improved if we change our font family order on the website. I don't think we should worry about how it displays in the issue header, as it is not designed to display code that way, although your screenshots are demonstrative.

**Comment by VojtechStep** [comment]
I tried to find out how the different fonts used for the website show the two symbols, and this is what I found (I think some of the fonts don't support the characters, in which case the browser probably chooses a fallback font):

- Menlo: no information, I don't own any Apple products
- Source Code Pro: indistinguishable [1]
- Consolas: indistinguishable [2]
- Monaco: indistinguishable [3]
- Lucida Console: distinct, bullet is a non-centered square [4]
- Liberation Mono: indistinguishable [5]
- DejaVu Sans Mono: distinct, bullet is a square [6]
- Bitstream Vera Sans Mono: indistinguishable [7]
- Courier New OS: distinct, but I would need to see them side-by-side [8]

source: fontmirror, fonts.com

I think having these two symbols be almost identical is more frequent then not. If I'm looking at the website's source correctly, it ships a copy of Source Code Pro; it would make sense to me to distribute also a font which distinguishes the two dots - from this list, that would be Lucida Console (but AFAIK that one needs a license, so that's a problem), or DejaVu Sans Mono (which should be redistributable).

1
![Source Code Pro](https://user-images.githubusercontent.com/15523887/219957726-73f5c155-8c59-43d6-b111-b6f94b1bdc1b.png)
2
![Consolas](https://user-images.githubusercontent.com/15523887/219957932-ed561c4b-d85f-48a0-a0db-f2a1ffaff89f.png)
3
![Monaco](https://user-images.githubusercontent.com/15523887/219957984-28c30a25-1613-44d6-98e6-a4142702c5b6.png)
4
![Lucida Console](https://user-images.githubusercontent.com/15523887/219958107-9d3ed955-5220-42c7-ae59-7011f5b33405.png)
5
![Liberation Mono](https://user-images.githubusercontent.com/15523887/219958180-794d3eb6-f8ea-4474-8f73-dce9a2f093ea.png)
6
![DejaVu Sans Mono](https://user-images.githubusercontent.com/15523887/219958241-08a5b4ec-36cb-44b9-8743-2da6f0f903eb.png)
7
![Bitstream Vera Sans Mono](https://user-images.githubusercontent.com/15523887/219958323-21c7fff6-7feb-4ddf-a9b5-337deea1fea8.png)
8
![Courier New OS](https://user-images.githubusercontent.com/15523887/219958484-627fbccf-9b40-493b-b699-187edef284ed.png)

**Comment by fredrik-bakke** [comment]
I agree with this. Another option would also be to redistribute Courier New.

**Comment by fredrik-bakke** [comment]
@jonaprieto Do you know how to fix this?

**Comment by fredrik-bakke** [comment]
For reference, here's how the symbols are displayed in my Brave browser on a mac:
**In agda code:**
![Screenshot 2023-02-19 at 17 17 36](https://user-images.githubusercontent.com/5176294/219960632-75b4d24b-a525-42db-809e-ea81fb36e1fb.png)
![Screenshot 2023-02-19 at 17 17 15](https://user-images.githubusercontent.com/5176294/219960635-636b9358-2911-4783-9bcd-e1be66cea1e7.png)
Looks like perhaps Deja Vu Mono is used here.

**In non-agda code:**
![Screenshot 2023-02-19 at 17 17 54](https://user-images.githubusercontent.com/5176294/219960664-3886fe8a-662a-45ad-b2d8-bbdc9a08b6c6.png)
![Screenshot 2023-02-19 at 17 18 05](https://user-images.githubusercontent.com/5176294/219960668-a12fb401-cc64-4b3c-bb68-8020d319f56d.png)

As an extension of this issue, it would make sense if the same font was used in both of these scenarios.

**Comment by VojtechStep** [comment]
FWIW to find out what font Brave (or other chromium-based browser) uses for a chunk of text, you can select the text, right click -> inspect -> switch the side pane from "Styles" to "Computed" and there you can find a list "Rendered fonts".
For me the Agda code blocks use "JetBrains Mono Medium", and inline code snippets use "Source Code Pro Medium".

**Comment by jonaprieto** [comment]
I'll assign myself to this issue. @fredrik-bakke ;)
